packages/react-link/src/components/Link/useLinkState.ts

```diff
@@ -25,38 +27,39 @@ export const useLinkState = (state: LinkState): LinkState => {
   // Adjust props depending on the root type.
   if (typeof as === 'string') {
     if (as === 'a') {
-      state.href = disabled ? undefined : href;
-      state.tabIndex = disabled && !disabledFocusable ? undefined : 0;
+      state.root;
+      state.root.href = disabled ? undefined : href;
+      state.root.tabIndex = disabled && !disabledFocusable ? undefined : 0;
     }
     // Remove the href, rel and target props for all non-anchor elements.
     else {
-      state.href = undefined;
-      state.rel = undefined;
-      state.target = undefined;
+      state.root.href = undefined;
+      state.root.rel = undefined;
+      state.root.target = undefined;
 
       // Add 'role=link' for all non-anchor elements.
-      state.role = 'link';
+      state.root.role = 'link';
```

Comment:
This overrides the `role` that the user might have passed in. Should you check if it's been set before changing it?

Or alternatively, this could be given a default value before spreading props; e.g.:
```
    root: getNativeElementProps(rootAs, {
      role: rootAs === 'a' ? undefined : 'link',
      ref,
      ...props,
    }),
```

Reply:
I decided to keep it in `useLinkState` as it is not a default in all cases.